python/pycylon/data/table.pyx

```diff
@@ -418,31 +581,58 @@ cdef class Table:
             ar_lst.append(npr)
         return np.array(ar_lst, order=order).T
 
+    def to_pydict(self):
+        """
+        creating a dictionary from PyCylon table
+        @return: dict
+        """
+        return self.to_arrow().to_pydict()
+
+    def to_csv(self, path, csv_write_options):
+        """
+        creating a csv file with PyCylon table data
+        @param path: str
+        @param csv_write_options: CSVWriteOptions
+        """
+        cdef string cpath = path.encode()
+        cdef CCSVWriteOptions c_csv_write_options = pycylon_unwrap_csv_write_options(
+            csv_write_options)
+        self.table_shd_ptr.get().WriteCSV(cpath, c_csv_write_options)
+
+    def to_arrow(self) -> pa.Table:
+        '''
+         creating PyArrow Table from PyCylon table
+         :param self: PyCylon Table
+         :return: PyArrow Table
+         '''
+        cdef shared_ptr[CArrowTable] converted_tb
+        cdef CStatus status = self.table_shd_ptr.get().ToArrowTable(converted_tb)
+        if status.is_ok():
+            return pyarrow_wrap_table(converted_tb)
+        else:
+            raise Exception("Table couldn't be converted to a PyArrow Table")
+
     @property
     def column_names(self):
-        table = to_pyarrow_table(self.id.encode())
-        py_arrow_table = pyarrow_wrap_table(table)
-        return py_arrow_table.column_names
+        """
+        Produces column names from PyCylon Table
+        @return: list
+        """
+        return self.to_arrow().column_names
 
     @property
     def schema(self):
```

Comment:
I feel like we'd have to distinguish that this schema returned is an `arrow schema`. @supunkamburugamuve WDYT?

Reply:
Maybe we need to add our own Schema class in CPP or I can do it easily in Python itself. 
What should be the best?

Reply:
If this is arrow schema, do we need to return it? Someone can get the arrow table and from there get the arrow schema?

Reply:
Agree. But our table also need a schema? We shouldn't encourage them to use arrow stuff in and out.

Reply:
I think if it is a sepearate schema lets do it in another pull request. Lets remove this method for now

Reply:
@supunkamburugamuve WDYT?